File: leetcode/JS/1877. Minimize Maximum Pair Sum in Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minPairSum = function (nums) {
  let arr = nums.sort((a, b) => a - b);
  let l = 0,
    r = nums.length - 1,
    res = 0;
  while (l < r) {
    res = Math.max(res, arr[l] + arr[r]);
    l++;
    r--;
  }

  return res;
};

File: leetcode/Python/1877. Minimize Maximum Pair Sum in Array.py
# class Solution:
#     def minPairSum(self, nums: List[int]) -> int:
#         l,r = 0, len(nums) - 1
#         ans = 0
#         nums.sort()
#         while l<r:
#             ans = max(ans, nums[l] + nums[r])
#             l += 1
#             r -= 1
            
#         return ans